temporal-test-server/src/test/java/io/temporal/testserver/functional/SignalLinksTest.java

```diff
@@ -0,0 +1,166 @@
+                                        .setWorkflowId("someWorkflow")
+                                        .setRunId(execution.getRunId())
+                                        .setNamespace("default")
+                                        .build())
+                        .build();
+
+        SignalWorkflowExecutionRequest signalRequest =
+                SignalWorkflowExecutionRequest.newBuilder()
+                        .setNamespace(testWorkflowRule.getWorkflowClient().getOptions().getNamespace())
+                        .setInput(Payloads.newBuilder().build())
+                        .setWorkflowExecution(execution)
+                        .setSignalName("test-signal")
+                        .addLinks(testLink)
+                        .build();
+
+        testWorkflowRule
+                .getWorkflowServiceStubs()
+                .blockingStub()
+                .signalWorkflowExecution(signalRequest);
+
+        stub.getResult(Void.class);
+
+        verifySignalLink(execution, testLink);
+    }
+
+    @Test
+    public void testSignalWithStartLinks() {
+        String workflowId = "test-workflow-id";
+        Link testLink =
+                Link.newBuilder()
+                        .setWorkflowEvent(
+                                Link.WorkflowEvent.newBuilder()
+                                        .setWorkflowId("someWorkflow")
+                                        .setRunId("some-run-id")
+                                        .setNamespace("default")
+                                        .build())
+                        .build();
+
+        SignalWithStartWorkflowExecutionRequest signalWithStartRequest =
+                SignalWithStartWorkflowExecutionRequest.newBuilder()
+                        .setTaskQueue(TaskQueue.newBuilder().setName(testWorkflowRule.getTaskQueue()).build())
+                        .setNamespace(testWorkflowRule.getWorkflowClient().getOptions().getNamespace())
+                        .setWorkflowType(WorkflowType.newBuilder().setName("TestWorkflow").build())
+                        .setSignalInput(Payloads.newBuilder().build())
+                        .setRequestId(randomUUID().toString())
+                        .setWorkflowId(workflowId)
+                        .setSignalName("test-signal")
+                        .addLinks(testLink)
+                        .build();
+
+        SignalWithStartWorkflowExecutionResponse response =
+                testWorkflowRule
+                        .getWorkflowServiceStubs()
+                        .blockingStub()
+                        .signalWithStartWorkflowExecution(signalWithStartRequest);
+
+        WorkflowExecution execution =
+                WorkflowExecution.newBuilder()
+                        .setWorkflowId(workflowId)
+                        .setRunId(response.getRunId())
+                        .build();
+
+        verifySignalLink(execution, testLink);
```

Comment:
Please verify that there's also a link on the WorkflowExecutionStarted event with signal-with-start.